src/assets/js/userswitch.js

```diff
@@ -0,0 +1,150 @@
+            if (form.elements[i].name === "") {
+                continue;
+            }
+            switch (form.elements[i].nodeName) {
+                case 'INPUT':
+                    switch (form.elements[i].type) {
+                        case 'text':
+                        case 'hidden':
+                        case 'password':
+                        case 'button':
+                        case 'reset':
+                        case 'submit':
+                            q.push(form.elements[i].name + "=" + encodeURIComponent(form.elements[i].value));
+                            break;
+                        case 'checkbox':
+                        case 'radio':
+                            if (form.elements[i].checked) {
+                                q.push(form.elements[i].name + "=" + encodeURIComponent(form.elements[i].value));
+                            }
+                            break;
+                        case 'file':
+                            break;
+                    }
+                    break;
+                case 'TEXTAREA':
+                    q.push(form.elements[i].name + "=" + encodeURIComponent(form.elements[i].value));
+                    break;
+                case 'SELECT':
+                    switch (form.elements[i].type) {
+                        case 'select-one':
+                            q.push(form.elements[i].name + "=" + encodeURIComponent(form.elements[i].value));
+                            break;
+                        case 'select-multiple':
+                            for (j = form.elements[i].options.length - 1; j >= 0; j = j - 1) {
+                                if (form.elements[i].options[j].selected) {
+                                    q.push(form.elements[i].name + "=" + encodeURIComponent(form.elements[i].options[j].value));
+                                }
+                            }
+                            break;
+                    }
+                    break;
+                case 'BUTTON':
+                    switch (form.elements[i].type) {
+                        case 'reset':
+                        case 'submit':
+                        case 'button':
+                            q.push(form.elements[i].name + "=" + encodeURIComponent(form.elements[i].value));
+                            break;
+                    }
+                    break;
+            }
+        }
+        return q.join("&");
+    };
+
+    var sendSetIdentity = function () {
+        var form = this;
+        var formData = serialize(form);
+        ajax({
+            url: form.action,
+            type: form.method,
+            data: formData,
+            success: function () {
+                window.top.location.reload();
+            },
+            error: function (data) {
+                form.yiiActiveForm('updateMessages', data.responseJSON, true); // TODO: jquery-less yii or yii-less form
```

Comment:
I'm afraid jQuery-less form is doable in Yii 3.0 only...

Reply:
Is this in the backend or toolbar?

Reply:
@schmunk42 it's used here: [http://demo.karlen.li/web/debug/default/view?panel=user&tag=5bf5177ae5f3c](http://demo.karlen.li/web/debug/default/view?panel=user&tag=5bf5177ae5f3c)